library/ui/src/main/java/com/google/android/exoplayer2/ui/PlayerControlView.java

```diff
@@ -835,13 +839,33 @@ private void updateProgress() {
         if (playbackSpeed <= 0.1f) {
```

Comment:
These test for speed <0.1 and speed > 5.0 where specifically chosen because all speed values between 0.1 and 5.0 would result in a delay between 200ms and 1000ms anyway. 

As you now limit to the delay in another way and based on other criteria, I think you should be able to just remove these extra tests and let all cases go through the central code.